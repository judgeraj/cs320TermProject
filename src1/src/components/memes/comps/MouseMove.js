import React, { Component } from "react";
import './memes.css';

class MouseMove extends Component {
  constructor(props) {
    super(props);

    this.state = {
      mouseMoving: false,
    };

    this.setMouseMove = this.setMouseMove.bind(this);
  }

  setMouseMove(e) {
    e.preventDefault();
    this.setState({ mouseMoving: true });

    let timeout;
    (() => {
      clearTimeout(timeout);
      timeout = setTimeout(() => this.setState({ mouseMoving: false }), 3000);
    })();
  }

  render() {
    const scrollButtonStyle = {
      visibility: this.state.mouseMoving ? "visible" : "hidden",
      //color: "green",
      backgroundColor: "green",
    };

    return (
      <div data-testid="mmove" onMouseMove={(e) => this.setMouseMove(e)}>
        <h1>Are you online?</h1>
        <button style={scrollButtonStyle}>You are online</button>
        &nbsp;&nbsp;&nbsp;
        <button style={scrollButtonStyle}>You are online</button>
        &nbsp;&nbsp;&nbsp;
        <button style={scrollButtonStyle}>You are online</button>
      </div>
    );
  }
}

export default MouseMove;
