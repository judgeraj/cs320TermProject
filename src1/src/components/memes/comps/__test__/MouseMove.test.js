import React from "react";
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import renderer from "react-test-renderer";
import MouseMove from "../MouseMove";


it("renders without problem", ()=>{
    const div = document.createElement("div");
    render(<MouseMove></MouseMove>, div)
})

it("render button without problem", ()=> {
    const {getByTestId} = render(<MouseMove></MouseMove>)
    expect(getByTestId('mmove')).toHaveTextContent("Are you online?")

})




