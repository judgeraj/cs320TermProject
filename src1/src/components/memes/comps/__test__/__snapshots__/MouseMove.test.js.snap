// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="mmove"
  onMouseMove={[Function]}
>
  <h1>
    Are you online?
  </h1>
  <button
    style={
      Object {
        "backgroundColor": "green",
        "visibility": "hidden",
      }
    }
  >
    You are online
  </button>
     
  <button
    style={
      Object {
        "backgroundColor": "green",
        "visibility": "hidden",
      }
    }
  >
    You are online
  </button>
     
  <button
    style={
      Object {
        "backgroundColor": "green",
        "visibility": "hidden",
      }
    }
  >
    You are online
  </button>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  data-testid="mmove"
  onMouseMove={[Function]}
>
  <h1>
    Are you online?
  </h1>
  <button
    style={
      Object {
        "backgroundColor": "green",
        "visibility": "hidden",
      }
    }
  >
    You are online
  </button>
     
  <button
    style={
      Object {
        "backgroundColor": "green",
        "visibility": "hidden",
      }
    }
  >
    You are online
  </button>
     
  <button
    style={
      Object {
        "backgroundColor": "green",
        "visibility": "hidden",
      }
    }
  >
    You are online
  </button>
</div>
`;
